docs/configuration.md

```diff
@@ -12,6 +12,24 @@ You can configure the `window.$docsify`.
 </script>
 ```
 
+The config can also be defined as a function, in which case the first arg is the Docsify `vm` instance. The function should return a config object. This can be useful for referencing `vm` in places like the markdown configuration:
+
+```html
+<script>
+  window.$docsify = function(vm) {
+    return {
+      markdown: {
+        renderer: {
+          code(code, lang) {
+            // ... use `vm` ...
+          },
+        },
+      },
+    };
+  };
+</script>
```

Comment:
Another approach we could take is to instead make sure we also pass in `vm` to all the markdown renderer hooks. But the current approach is easier and more resilient to changes in the code where marked is being called.

Reply:
wrapper function for hooks [already has the `vm`](https://docsify.js.org/#/write-a-plugin?id=full-configuration) as a parameter.

Reply:
Those are for Docsify hooks, but not for `marked` renderer plugins.

Reply:
make sense